yunwoo/programmers-1/문자열-나누기.js

```diff
@@ -0,0 +1,30 @@
+function solution(s) {
+  let copyString = s;
+  let result = 0;
+  let matchChar = 0;
+  let misMatchChar = 0;
+
+  for (let i = 0; i <= copyString.length; i++) {
+    copyString[i] === copyString[0] ? matchChar++ : misMatchChar++;
+
+    if (matchChar === misMatchChar) {
+      copyString = copyString.replace(copyString.substring(0, i + 1), "");
+      i = -1;
+      matchChar = 0;
+      misMatchChar = 0;
+      result++;
+    }
+  }
+
+  return misMatchChar ? result + 1 : result;
+}
```

Comment:
```suggestion
function solution(s) {
  let copyString = s;
  let result = 0;
  let matchCount = 0;
  let misMatchCount = 0;

  for (let i = 0; i <= copyString.length; i++) {
    copyString[i] === copyString[0] ? matchCount++ : misMatchCount++;

    if (matchCount === misMatchCount) {
      copyString = copyString.replace(copyString.substring(0, i + 1), "");
      i = -1;
      matchCount = 0;
      misMatchCount = 0;
      result++;
    }
  }

  return misMatchCount ? result + 1 : result;
}
```
- replace, substring을 이용한 풀이 잘 봤습니다 🙇‍♂️
- matchChar, misMatchChar 모두 char라는 변수의 의미는 단어라는 뜻이기 때문에 현재 풀이에서는 카운팅을 목적으로 하고있어서 count가 더 적절한 것 같습니다.
- misMathc인 경우는 `abracadabra` 인 경우 ab, ra, ca, da, br씩 나뉘다가 마지막에 a를 추가해야되는데, 해당 a 부분이 문제에서 
```
만약 두 횟수가 다른 상태에서 더 이상 읽을 글자가 없다면, 역시 지금까지 읽은 문자열을 분리하고, 종료합니다.
```
부분입니다. 해당 문자열도 마지막에 결국 분리되서 result로 추가해야되기 때문입니다.

Reply:
- 아..! 변수명 더 잘 지어 보겠습니다!!🙇‍♂️
- 나름 문제를 꼼꼼히 읽어봤다고 생각했는데 갈수록 어렵네요 🥲 이해했습니다 감사합니다 👍👍👍